docs/source/notes/extending.rst

```diff
@@ -644,7 +644,7 @@ implementation more permissive about what operations are allowed::
           if kwargs is None:
               kwargs = {}
           args = [a._t if hasattr(a, '_t') else a for a in args]
-          metadatas = tuple(a._metadata if hasattr(a, '_metadata') for a in args)
+          metadatas = tuple(a._metadata for a in args if hasattr(a, '_metadata'))
```

Comment:
@pmeier @albanD actually even with this fix example isn't working as we always hit `assert len(metadatas) > 0`:
```python
args = [a._t if hasattr(a, '_t') else a for a in args]
# Now args contains torch.Tensor which do not have any metadata
metadatas = tuple(a._metadata for a in args if hasattr(a, '_metadata'))
assert len(metadatas) > 0
```
I think it should be like that:
```python
metadatas = tuple(a._metadata for a in args if hasattr(a, '_metadata'))
args = [a._t if hasattr(a, '_t') else a for a in args]
```